src/Common.proto

```diff
@@ -33,6 +33,16 @@ message PeerId {
     bytes publicKey = 5; //20 bytes
 }
 
+/**
+* Provides a network error message, for when a node cant send/ build a valid response
+*/
+message ProtocolErrorMessageSigned {
```

Comment:
If it followed the same structure as the other messages in this file, wouldn't we have ProtocolErrorMessage and then ProtocolErrorMessageSigned (containing just signature and ProtocolErrorMessage fields)?

Reply:
mmmm I only see this being used in handlers when an exception is raised it may not be propagated through the entire pipeline in the same way a standard network message is and being directed to a single error handler, but maybe it should?

Reply:
I'm happy to sign off on this if you decide this is what you want, just thought it was worth mentioning.